### ISSUE TITLE
[java] DanglingJavadoc: False positive on /// comments for Java < 23

### ISSUE THREAD

**Issue by mebigfatguy** [pmd/pmd#6450]
PMD  7.21.0

```java
public class JD {

    public void test() {

        //// A comment
        while (true);
    }
}
```
This is not javadoc, so no javadoc rule infringement should be produced.

/home/dave/sample/JD.java:6:	DanglingJavadoc:	Javadoc comment does not belong to any class, method or field

**Running PMD through:** *[CLI]*

**Comment by adangel** [comment]
Thanks for reporting.

Note, since Java 23, three slashes start a markdown javadoc comment - see [JEP 467: Markdown Documentation Comments](https://openjdk.org/jeps/467).

I'v just tested it with Java 25, and even with your four slashes, it ends up as javadoc:

This code

```java
//// Test Class
public class Test {}
```

Produces this Javadoc:

<img width="316" height="361" alt="Image" src="https://github.com/user-attachments/assets/cb829c7a-7431-4a79-8a04-b02800f9c9b7" />

Interestingly, the slash ends up in the documentation (which could be considered a bug in the JDK)...

Regarding to PMD: If you call PMD and select a java version < 23, then ideally, the rule shouldn't report a violation. But for java versions >= 23, it should, as this is a valid documentation comment.

Did you select a specific java version via [--use-version](https://docs.pmd-code.org/latest/pmd_userdocs_cli_reference.html#-use-version)?

Since the first slash of the four slashes actually end up in the documentation, this is probably not what you want. We could create a new rule "InvalidDocumentationComment" for such cases.
Just use the normal two slashes for a comment followed by a space, like `// // A comment` to be on the safe side.

**Comment by zbynek** [comment]
@adangel the handling of `///` as comment in Java <23 is something we considered and decided to keep simple:
https://github.com/pmd/pmd/pull/5916#pullrequestreview-3209399308
If we were to fix it, should it be for this rule specifically or should we change the handling of `///` in general?

> Interestingly, the slash ends up in the documentation (which could be considered a bug in the JDK)

I think that's fine, three slashes indicate JavaDoc, fourth slash is part of the comment. Maybe IDEs could highlight it differently to make it more obvious.

**Comment by victorwss** [comment]
Well, I would argue that a `///` comment in a pre-23 Java version is a problem that should be flagged nonetheless because code featuring this would likely break the Javadoc in Java 23+ and hence, it is a problem for a modernization/migration effort. In fact, I would be concerned with the opposite problem: Having `///` javadocs in pre-23 Java.

**Comment by UncleOwen** [comment]
> Well, I would argue that a `///` comment in a pre-23 Java version is a problem that should be flagged nonetheless because code featuring this would likely break the Javadoc in Java 23+ and hence, it is a problem for a modernization/migration effort. In fact, I would be concerned with the opposite problem: Having `///` javadocs in pre-23 Java.

That's not what this rule is about. But sounds like a good idea for another rule.
